### PR TITLE
feat(simulation): check whether the op fits within a valid time range

### PR DESCRIPTION
**Pull request by 0xfourzerofour**
## Proposed Changes

  - Add check during simulation for a valid time range and add a 15 second buffer

### CODE REVIEW

File: crates/types/src/pool/error.rs [hunk 1]
```diff
@@ -152,6 +152,9 @@ pub enum SimulationViolation {
     /// The signature is invalid for the account
     #[display("invalid account signature")]
     InvalidAccountSignature,
+    /// The user operation has an invalid time range based on the `valid_until` and `valid_after` fields
+    #[display("User Operation expired or has an invalid time range")]
+    InvalidTimeRange,
```
Comment (dancoombs) [inline crates/types/src/pool/error.rs:157]: Can we return `valid_until` and `valid_after` as part of this error? Can put it in the `data` field of the RPC error

Comment (0xfourzerofour) [reply crates/types/src/pool/error.rs:157]: Yeh I can add that

File: crates/types/src/validation_results.rs [hunk 2]
```diff
@@ -259,6 +261,17 @@ pub struct ValidationReturnInfo {
     pub paymaster_context: Bytes,
 }
 
+const VALID_UNTIL_FUTURE_SECONDS: u64 = 15;
```
Comment (dancoombs) [inline crates/types/src/validation_results.rs:264]: I thought I remember us having something similar to this elsewhere in the codebase. Are we checking these values elsewhere? Maybe in the builder?

Comment (dancoombs) [reply crates/types/src/validation_results.rs:264]: Yeah found it https://github.com/alchemyplatform/rundler/blob/02af86989ec90045bbee7e2890062128665adf3b/crates/builder/src/bundle_proposer.rs#L53

Comment (dancoombs) [reply crates/types/src/validation_results.rs:264]: Also we have a function that does this buffer logic: https://github.com/alchemyplatform/rundler/blob/02af86989ec90045bbee7e2890062128665adf3b/crates/builder/src/bundle_proposer.rs#L447

Comment (dancoombs) [reply crates/types/src/validation_results.rs:264]: Lets consolidate this somewhere

Comment (0xfourzerofour) [reply crates/types/src/validation_results.rs]: I added it to the userop types file and import it into both

**Review by dancoombs — APPROVED** [review]
LGTM!

We should add to our docs that we require a UO to be valid at least 1 minute in the future for acceptance into the mempool. This is important as an anti-DOS mechanism, we don't want to fill our mempool with UOs that are going to expire before bundling.

**Review by dancoombs — APPROVED** [review]

**Pull request merged**